test/jpypetest/test_array.py

```diff
@@ -176,3 +173,184 @@ def testJArrayCopyRange(self):
+
+    @common.requireNumpy
+    def testArrayOfFloat(self):
+        self.checkArrayOf(JFloat, np.float32)
+
+    @common.requireNumpy
+    def testArrayOfDouble(self):
+        self.checkArrayOf(JDouble, np.float64)
+
+    @common.requireNumpy
+    def testArrayOfBooleanCast(self):
+        self.checkArrayOfCast(JBoolean, np.bool)
+
+    @common.requireNumpy
+    def testArrayOfByteCast(self):
+        self.checkArrayOfCast(JByte, np.int8)
+
+    @common.requireNumpy
+    def testArrayOfShortCast(self):
+        self.checkArrayOfCast(JShort, np.int16)
+
+    @common.requireNumpy
+    def testArrayOfIntCast(self):
+        self.checkArrayOfCast(JInt, np.int32)
+
+    @common.requireNumpy
+    def testArrayOfLongCast(self):
+        self.checkArrayOfCast(JLong, np.int64)
+
+    @common.requireNumpy
+    def testArrayOfFloatCast(self):
+        self.checkArrayOfCast(JFloat, np.float32)
+
+    @common.requireNumpy
+    def testArrayOfDoubleCast(self):
+        self.checkArrayOfCast(JDouble, np.float64)
+
+    def testArrayOfExc(self):
+        a = np.random.randint(0, 2*8-1, size=1, dtype=np.uint8)
+        with self.assertRaises(TypeError):
+            JArray.of(a)
+        a = np.random.randint(0, 2*16-1, size=1, dtype=np.uint16)
+        with self.assertRaises(TypeError):
+            JArray.of(a)
+        a = np.random.randint(0, 2*32-1, size=1, dtype=np.uint32)
+        with self.assertRaises(TypeError):
+            JArray.of(a)
+        a = np.random.randint(0, 2*64-1, size=1, dtype=np.uint64)
+        with self.assertRaises(TypeError):
+            JArray.of(a)
+
+    @common.requireInstrumentation
+    def testArrayOfFaults(self):
+        b = bytes([1,2,3])
+        _jpype.fault("JPJavaFrame::assemble")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            JArray.of(b, JInt)
+        _jpype.fault("JPBooleanType::newMultiArray")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            JArray.of(b, JBoolean)
+        _jpype.fault("JPCharType::newMultiArray")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            JArray.of(b, JChar)
+        _jpype.fault("JPByteType::newMultiArray")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            JArray.of(b, JByte)
+        _jpype.fault("JPShortType::newMultiArray")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            JArray.of(b, JShort)
+        _jpype.fault("JPIntType::newMultiArray")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            JArray.of(b, JInt)
+        _jpype.fault("JPLongType::newMultiArray")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            JArray.of(b, JLong)
+        _jpype.fault("JPFloatType::newMultiArray")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            JArray.of(b, JFloat)
+        _jpype.fault("JPDoubleType::newMultiArray")
+        with self.assertRaisesRegex(SystemError, "fault"):
+            JArray.of(b, JDouble)
+ 
```

Comment:
I cannot spot a test for the handling of memory fragmented (stridded) arrays.

Reply:
I need to find a way to test fortran and strided arrays.  Both are sources of potential bugs.  Any idea how those should be tested?

Reply:
what is the intended result of such a memory transfer? You could check, that a strided array created with numpy survives the transfer to Java and back to Python.

Reply:
Is there a way to create a strided array in numpy?  I was able to create a normal array with different steps by slincing and a c contigous by disabling to hit the fall through.  I just dont know how to hit the others currently.  

Perhaps i should look through numpy tests and see.  Or maybe we need a cpython harness?

Reply:
I think this function can become handy in this case, https://docs.scipy.org/doc/numpy/reference/generated/numpy.lib.stride_tricks.as_strided.html

Reply:
Okay I will check it out and see if we can complete the test bench for this.  Many thanks.